core/src/main/java/com/netflix/conductor/validations/WorkflowTaskTypeConstraint.java

```diff
@@ -98,11 +98,43 @@ public boolean isValid(WorkflowTask workflowTask, ConstraintValidatorContext con
                 case TaskType.TASK_TYPE_JSON_JQ_TRANSFORM:
                     valid = isJSONJQTransformTaskValid(workflowTask, context);
                     break;
+                case TaskType.TASK_TYPE_SIMPLE:
+                    valid = isSimpleTaskValid(workflowTask, context);
+                    break;
             }
 
             return valid;
         }
 
+        private boolean isSimpleTaskValid(WorkflowTask workflowTask, ConstraintValidatorContext context) {
+            TaskDef task = workflowTask.getTaskDefinition();
+            if(task == null) {
+                task = ValidationContext.getMetadataDAO().getTaskDef(workflowTask.getName());
+            }
+            if (task == null) {
+                if( workflowTask.getGlobalConcurrentExecutionLimit() + workflowTask.getLocalConcurrentExecutionLimit() != 0) {
+                    String message = String.format("workflowTask: %s must use an existing taskDef if *ConcurrentExecutionLimit is defined in the workflowTask.", workflowTask.getTaskReferenceName());
+                    context.buildConstraintViolationWithTemplate(message).addConstraintViolation();
+                }
+                return false;
+            }
+            boolean valid = true;
+            if (workflowTask.getGlobalConcurrentExecutionLimit() + workflowTask.getLocalConcurrentExecutionLimit() != 0
+                && task.getTimeoutSeconds() < 1)
```

Comment:
This verifies that when enabling the local or global concurrent execution limit, the linked TaskDef has a timeout defined for in the semaphore. The semaphore requires a timeout parameter to ensure that no dead Taskld s are holding the semaphore.

Reply:
@apanicker-nflx
I understand. 
What do you think about adding another string field to use as the semaphore name instead of the `taskReferenceName`? How about `concurrentLockName` or `concurrentSemaphoreName`?